### PR TITLE
Improve year validation

### PR DESCRIPTION
**Pull request by NicolaasDieBaas**
add parameters to is_valid_year function that allows you to pass in min/max

### CODE REVIEW

**Comment by NicolaasDieBaas** [comment]
@gsvr please review

File: src/utils.js [hunk 1]
```diff
@@ -123,9 +123,13 @@ go.utils = {
         return moment(date, format, true).isValid();
     },
 
-    is_valid_year: function(input) {
+    is_valid_year: function(input, min, max) {  // expecting string parameters
```
Comment (gsvr) [inline src/utils.js]: Let's call the variables `year`, `minYear` and `maxYear` for clarity

File: src/utils.js [hunk 2]
```diff
@@ -123,9 +123,13 @@ go.utils = {
...
         // check that it is a number and has four digits
-        return input.length === 4 && go.utils.check_valid_number(input);
+        // AND that the number is within the range determined by the min/max parameters
+        return input.length === 4 && min.length === 4 && max.length === 4 &&
```
Comment (gsvr) [inline src/utils.js]: I think we can drop this requirement now that we're using the minYear and maxYear fields.

**Comment by gsvr** [comment]
@NicolaasDieBaas First pass done.

**Comment by NicolaasDieBaas** [comment]
@gsvr 2nd pass plz

File: src/utils.js [hunk 3]
```diff
@@ -123,9 +123,11 @@ go.utils = {
         return moment(date, format, true).isValid();
     },
 
-    is_valid_year: function(input) {
-        // check that it is a number and has four digits
-        return input.length === 4 && go.utils.check_valid_number(input);
+    is_valid_year: function(year, minYear, maxYear) {  // expecting string parameters
+        // check that the number is within the range determined by the minYear/maxYear parameters
+        return go.utils.check_valid_number(year) &&
+                parseInt(year, 10) >= parseInt(minYear, 10) &&
```
Comment (gsvr) [inline src/utils.js]: indentation

**Comment by gsvr** [comment]
:+1:

**Pull request merged**